### DISCUSSION TITLE
Validation failed: Expected DataObject\Data\ObjectMetadata

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#7713] [Q&A / Support] [answered]
We use  AdvancedManyToManyObjectRelation into my class and save data from controller dynamically and get error because of I have added extra column into AdvancedManyToManyObjectRelation field.
![image](https://user-images.githubusercontent.com/52573912/101741577-3c1c9400-3af0-11eb-8500-fa81103940d9.png)

Any idea how I can handle it.
Thanks.

**Comment by dvesh3** [comment, marked as answer]
@rushit-credencys could you please share code example of how you are adding relations to AdvancedManyToManyObjectRelation field? thanks!

please make sure you follow the example here https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Relation_Types.html#page_Save-objects-with-metadata

**Reply by ghost** [reply]
Ya, Thanks @dvesh3 it's working.
